artifacts/deployment/etcdproxy-controller.yaml

```diff
@@ -0,0 +1,121 @@
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: kube-apiserver-storage
+spec:
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: etcdproxy-sa
+  namespace: kube-apiserver-storage
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: etcdproxy-crd-clusterrole
+rules:
+- apiGroups: ["etcd.xmudrii.com"]
+  resources: ["etcdstorages"]
+  verbs: ["get", "watch", "list"]
+- apiGroups: ["etcd.xmudrii.com"]
+  resources: ["etcdstorages/status"]
+  verbs: ["get", "watch", "list", "create", "update", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: etcdproxy-role
+  namespace: kube-apiserver-storage
+rules:
```

Comment:
The ServiceAccount for the controller only needs access to the ReplicaSets and Services.

The ServiceAccount for the etcd proxy needs read-only access to core etcd ConfigMaps/Secrets, and read/write access to other secrets/configmaps (used by etcd proxies).